### PR TITLE
Automate cargo crate publish process 

### PR DESCRIPTION
**Pull request by gdhuper**
# Description

- Added CI pipeline to automate cargo crate publish on crates.io 
- Updated README with instructions on how to use published crate

## Issue reference

This PR will close: #33 

## Checklist

* [x] Code compiles correctly
* [x] Extended the documentation
* [x] Crate Versioning Scheme 
* [x] Update github-actions url for README build badge

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,14 @@
 [package]
 name = "dapr"
-version = "0.2.0"
+version = "1.0.0"
```
Comment (rylev) [inline Cargo.toml:3]: 1.0 seems pretty aggressive. I was under the impression that this crate was still very much a work in progress.

Comment (gdhuper) [reply Cargo.toml:3]: Agree. Should this match the version for dapr runtime?

**Comment by gdhuper** [comment]
@youngbupark is there a versioning scheme that you follow for sdk versions?

**Comment by youngbupark** [comment]
> @youngbupark is there a versioning scheme that you follow for sdk versions?

https://crates.io/crates/dapr

"0.1.0-alpha.2" is the last published version. `0.2.0-alpha.0` makes sense for this release.

/cc @yaron2 @msfussell

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - name: Build examples
+      run: cargo build --examples
+    - name: Run Tests
+      run: cargo test --all-targets
+   
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    needs: [lint, build]
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```
Comment (youngbupark) [inline .github/workflows/ci.yml:39]: I would publish the cargo package only for tag push. Please see the below command:

https://github.com/dapr/cli/blob/51bbd0a1715eb00a2cc6e3dd2518a42336418389/.github/workflows/dapr_cli.yaml#L89

So the release process (v0.2.0-alpha.0) will be 
1. create release-* branch
2. push v* tag with v0.2.0-alpha.0 

Please change this workflow to publish cargo package when it is version tag push.

**Review by youngbupark — CHANGES_REQUESTED** [review]
Thanks for this contribution! I left small comment.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (gdhuper) [reply .github/workflows/ci.yml:39]: Sounds good. This workflow makes more sense.

Comment (youngbupark) [reply .github/workflows/ci.yml:39]: @gdhuper any updates ?

Comment (gdhuper) [reply .github/workflows/ci.yml:39]: Apologies for delay. Will try to wrap this up end of today.

**Review by youngbupark — APPROVED** [review]
Thanks a lot!

**Pull request merged**